File: Mirror/py_proj/add_track.py
from PyQt5.QtWidgets import QDialog, QInputDialog, QFileDialog
from add_track_ui import *

class Add_track(QDialog):
    def __init__(self,parent):
        super().__init__()
        self.ui = Ui_Add_track()
        self.ui.setupUi(self)
        self.ui.pushButton.clicked.connect(self.on_pushButton_clicked)

    def on_pushButton_clicked(self):
        print('browse')
        fname, ok = QFileDialog.getOpenFileName(self, 'Open File','/home', 'MIDI files (*.mid)')
        if ok:
            self.training_file = fname
            self.ui.file_label.setText(fname)

File: cpp_mirror/add_track.cpp
#include "add_track.h"
#include "ui_add_track.h"

Add_track::Add_track(QWidget *parent) :
    QDialog(parent),
    ui(new Ui::Add_track)
{
    ui->setupUi(this);
}

Add_track::~Add_track()
{
    delete ui;
}

void Add_track::on_pushButton_clicked()
{
    training_file = QFileDialog::getOpenFileName(this, tr("Open File"),
                                                 "/home",
                                                 tr("MIDI files (*.mid)"));
    ui->file_label->setText(training_file);
}
